src/annotations/index.ts

```diff
@@ -345,13 +182,36 @@ export function deprecated(throws: boolean = true, context: Context = globalCont
     }
 }
 
-export function defineTypes(
-    target: typeof Schema,
-    fields: { [property: string]: DefinitionType },
-    context: Context = target._context || globalContext
-) {
-    for (let field in fields) {
-        type(fields[field], context)(target.prototype, field);
+export function hasFilter(klass: typeof Schema) {
+    return klass._context && klass._context.useFilters;
+}
+
+function applyFilter(addFilter:(definition: SchemaDefinition, field: string) => boolean) {
```

Comment:
codeclimate flagged filter and filterChildren as potentially duplicate code so I refactored to appease it.